src/Compilers/VisualBasic/Portable/BoundTree/Expression.vb

```diff
@@ -1382,6 +1370,105 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
         End Function
     End Class
 
+    Partial Class BoundFieldOrPropertyInitializer
+        Implements IFieldInitializer, IPropertyInitializer
```

Comment:
Some devs have a habit of doing type checks instead of kind checks. I think objects that we expose shouldn't implement interfaces in this fashion.